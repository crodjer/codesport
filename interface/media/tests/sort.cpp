#include<stdio.h>
#include<stdlib.h>
#include<math.h>
#include<string.h>

#define MAXNO 100 
#define EXCH(X,Y,Z) ((Z)=(X), (X)=(Y), (Y)=(Z))
void selectionSort(int [], int);
int main() // main.c
{
    int no , i ;
    int data[MAXNO] ;
 
    scanf("%d",&no);
    for(i=0;i<no;i++)
       scanf("%d", &data[i]);
    selectionSort(data, no) ;
    for(i = 0; i < no; ++i)
    {
      printf("%d\t", data[i]);
    }
    return 0 ;
}

void selectionSort(int data[], int nod)
 { 
     int i ;
     		
     for(i = 0; i < nod - 1; ++i) {
         int max, j ;
         int temp ;

         temp = data[i] ;
         max = i ;
         for(j = i+1; j < nod; ++j)
             if(data[j] > temp) {
                temp = data[j] ;
                max = j ;
             }
         EXCH(data[i], data[max], temp);
     }
}

