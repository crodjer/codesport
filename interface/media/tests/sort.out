10	10	10	10	10	10	10	10	10	10	7	7	7	7	7	7	7	7	7	7	4	4	4	4	4	4	4	4	4	4	3	3	3	3	3	3	3	3	3	3	1	1	1	1	1	1	1	1	1	1	
